File: frontend/src/services/dashboardService.ts
import api from './api';
import type { RecentTransaction } from "../types";

export const getSummary = async () => {
  const res = await api.get('/dashboard/summary');
  return res.data;
};

export const getTopItems = async () => {
  const res = await api.get('/dashboard/top-items');
  return res.data;
};

export const getRevenue = async (range: string) => {
  const res = await api.get(`/dashboard/revenue?range=${range}`);
  return res.data;
};

// export const getRecentTransactions = async () => {
//   const res = await api.get('/dashboard/recent-transactions');
//   return res.data;
// };

export const getRecentTransactions = async (): Promise<RecentTransaction[]> => {
  const res = await api.get('/dashboard/recent-transactions');
  return res.data;
};
export const getRecentTransactionsPaginated = async (page: number = 1, size: number = 10) => {
  const res = await api.get(`/dashboard/recent-transactions/paginated?page=${page}&size=${size}`);
  return res.data;
};
export const getRecentTransactionsWithFilters = async (
  page: number = 1, 
  size: number = 10,
  filters: {
    cashier?: string;
    minValue?: number;
    maxValue?: number;
    startDate?: string;
    endDate?: string;
    paymentMode?: string;
  }
) => {
  const params = new URLSearchParams();
  params.append('page', page.toString());
  params.append('size', size.toString());
  
  if (filters.cashier) params.append('cashier', filters.cashier);
  if (filters.minValue) params.append('minValue', filters.minValue.toString());
  if (filters.maxValue) params.append('maxValue', filters.maxValue.toString());
  if (filters.startDate) params.append('startDate', filters.startDate);
  if (filters.endDate) params.append('endDate', filters.endDate);
  if (filters.paymentMode) params.append('paymentMode', filters.paymentMode);
  
  const res = await api.get(`/dashboard/recent-transactions/filtered?${params.toString()}`);
  return res.data;
};

export const getAllCashiers = async () => {
  const res = await api.get('/dashboard/cashiers');
  return res.data;
};
export const getUsersPerformance = async (range: string) => {

  const res = await api.get(`/dashboard/users-performance?range=${range}`);
  return res.data;
};

export const getTodaysRevenueByPaymentMode = async () => {
  const res = await api.get('/dashboard/todays-revenue-by-payment-mode');
  return res.data;
};

export default { getSummary, getTopItems, getRevenue, getRecentTransactions, getUsersPerformance, getTodaysRevenueByPaymentMode, getRecentTransactionsPaginated, getRecentTransactionsWithFilters, getAllCashiers };

File: backend/cr-cafe-billing-core/src/main/java/com/crcafe/core/service/DashboardService.java
package com.crcafe.core.service;

import com.crcafe.core.dto.DashboardSummaryDto;
import com.crcafe.core.repository.UserOrderCountProjection;

import java.util.List;
import java.util.Map;

public interface DashboardService {
    DashboardSummaryDto getSummary();
    List<Map<String, Object>> getTopItems();
    List<Map<String, Object>> getRevenue(String range);
    List<Map<String, Object>> getRecentTransactions();
    Map<String, Object> getRecentTransactionsPaginated(int page, int size);
    Map<String, Object> getRecentTransactionsWithFilters(int page, int size, String cashier, 
        java.math.BigDecimal minValue, java.math.BigDecimal maxValue, 
        java.time.LocalDateTime startDate, java.time.LocalDateTime endDate, String paymentMode);
    List<String> getAllCashiers();
    List<Map<String,Object>> getUsersPerformance(String range);
    // Add method to get today's revenue by payment mode
    Map<String, java.math.BigDecimal> getTodaysRevenueByPaymentMode();
}